include/RevMem.h

```diff
@@ -445,6 +469,22 @@ class RevMem {
     return memStatsTotal;
   }
 
+  /// RevMem: Dump the memory contents
+  void DumpMem( const uint64_t startAddr,
+                const uint64_t numBytes,
+                const uint64_t bytesPerRow  = 16,
+                std::ostream&  outputStream = std::cout );
+
+  void DumpValidMem( const uint64_t bytesPerRow  = 16,
+                     std::ostream&  outputStream = std::cout );
+
+  void DumpMemSeg( std::shared_ptr< MemSegment > MemSeg,
```

Comment:
Unless a multithreaded Rev can destroy a `std::shared_ptr<MemSegment>` in another thread and you need to "borrow" `MemSeg` in a thread-safe way, you can use `const std::shared_ptr<MemSegment>& MemSeg` to be faster (it avoids atomic increment/decrement of reference count).